### PR TITLE
Decompile battle_dome

### PR DESCRIPTION
**Pull request by DizzyEggg**
This file will take a while.

### CODE REVIEW

File: src/battle_dome.c [hunk 1]
```diff
@@ -0,0 +1,6480 @@
...
+    {.src = 0x6021, .y = 0x06, .x = 0x09},
+    {.src = 0x6021, .y = 0x06, .x = 0x0a},
+    {.src = 0x6027, .y = 0x05, .x = 0x0a},
+    {.src = 0x6023, .y = 0x05, .x = 0x0b},
+    {.src = 0x6027, .y = 0x06, .x = 0x0b},
+    {.src = 0x6047, .y = 0x07, .x = 0x0b},
+    {.src = 0x6023, .y = 0x07, .x = 0x0c},
+    {.src = 0x6027, .y = 0x08, .x = 0x0c},
+    {.src = 0x6027, .y = 0x09, .x = 0x0c},
+    {.s
```
Comment (ProjectRevoTPP) [inline src/battle_dome.c]: tabs

Comment (ProjectRevoTPP) [inline src/battle_dome.c]: This isn't a bug. It's calculating the total points for all 3 mons on each side. Read the code more carefully.

Comment (DizzyEggg) [reply src/battle_dome.c]: Sadly, it's a bug. It should be cleared once the match's result is determined.

File: src/battle_dome.c [hunk 2]
```diff
@@ -0,0 +1,6487 @@
...
+    {.src = 0x6021, .y = 0x06, .x = 0x09},
+    {.src = 0x6021, .y = 0x06, .x = 0x0a},
+    {.src = 0x6027, .y = 0x05, .x = 0x0a},
+    {.src = 0x6023, .y = 0x05, .x = 0x0b},
+    {.src = 0x6027, .y = 0x06, .x = 0x0b},
+    {.src = 0x6047, .y = 0x07, .x = 0x0b},
+    {.src = 0x6023, .y = 0x07, .x = 0x0c},
+    {.src = 0x6027, .y = 0x08, .x = 0x0c},
+    {.src = 0x6027, .y = 0x09, .x = 0x0c},
+    {.src = 0x6027, .y = 0x0a, .x = 0x
```
Comment (Diegoisawesome) [inline src/battle_dome.c]: What's the significance of the number 510? Should that be some define, maybe?

Comment (Diegoisawesome) [inline src/battle_dome.c]: 5579: Might be 2D arrays

Comment (Diegoisawesome) [inline src/battle_dome.c]: 5981: RGB_BLACK

Comment (Diegoisawesome) [inline src/battle_dome.c]: 5933: RGB_BLACK

Comment (Diegoisawesome) [inline src/battle_dome.c]: 5415, 5429, 5435, 5439: RGB_BLACK

Comment (Diegoisawesome) [inline src/battle_dome.c]: 3936, 3949: RGB_BLACK

Comment (Diegoisawesome) [inline src/battle_dome.c]: HblankCb_BattleDome: Use macros for WININ/OUT/etc

Comment (DizzyEggg) [reply src/battle_dome.c]: Think twice.
Hint, check out the bestId var:P

Comment (Diegoisawesome) [reply src/battle_dome.c]: Ah, true.

**Pull request merged**